### PR TITLE
Add qml.counts UI

### PR DESCRIPTION
**Pull request by AlbertMitjans**
**Context:**  Add `qml.counts` UI feature.

**Description of the Change:** Replace `qml.sample(counts=True)` by `qml.counts`. Change tests accordingly.

### CODE REVIEW

**Comment by AlbertMitjans** [comment]
[sc-23807]

File: tests/test_measurements.py [hunk 1]
```diff
@@ -391,6 +390,22 @@ def circuit():
...
 
+    def test_providing_symbolic_operation_raises_error(self):
+        """Test that a qml.QuantumFunctionError is raised if the provided
+        argument is a symbolic operation"""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.52, wires=0)
+            return qml.sample(qml.op_sum(qml.PauliX(0), qml.PauliZ(0)))
```
Comment (antalszava) [inline tests/test_measurements.py:401]: Was this test pulled in from another branch? This seems to be testing `qml.sample`.

**Review by antalszava — APPROVED** [review]
Looks good @AlbertMitjans! :tada: Thank you for looking into this :heart_eyes: 

Was curious about the symbolic op tests.

**Comment by antalszava** [comment]
The failures seem to be due to having a new `tests/test_new_return_types.py` file that contained the old version of the counts UI.

File: pennylane/measurements.py [hunk 2]
```diff
@@ -185,7 +185,8 @@ def numeric_type(self):
             return numeric_type
 
         raise qml.QuantumFunctionError(
-            f"Cannot deduce the numeric type of the measurement process with unrecognized return_type {self.return_type}."
+            "Cannot deduce the numeric type of the measurement process with unrecognized "
+            + f"return_type {self.return_type}."
```
[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/measurements.py:189]: This change was just to avoid a `line too long` warning.

File: tests/test_measurements.py (hunk 1 above)
Comment (AlbertMitjans) [reply tests/test_measurements.py:401]: This test was added for both `qml.sample` and `qml.count` to cover the line (which is now present in both functions):
https://github.com/PennyLaneAI/pennylane/blob/1558f7d561b0cc709630e2f9d030101afa873b36/pennylane/measurements.py#L686

Comment (AlbertMitjans) [reply tests/test_measurements.py:401]: I just realised that this line is covered in `test_sum.py` and `test_sprod.py`:
https://github.com/PennyLaneAI/pennylane/blob/1558f7d561b0cc709630e2f9d030101afa873b36/tests/ops/op_math/test_sum.py#L695
I will add another test to cover the error for the `qml.count` method.

**Pull request merged**